### PR TITLE
Compiling TC under FreeBSD 10.x

### PR DESCRIPTION
**Pull request by snakkur**

### CODE REVIEW

File: dep/g3dlite/include/G3D/System.h [hunk 1]
```diff
@@ -21,6 +21,10 @@
 #include "G3D/FileNotFound.h"
 #include <string>
 
+#ifdef G3D_FREEBSD
+#   include <sys/_timeval.h>
```
Comment (ijustwantedtosayhi) [inline dep/g3dlite/include/G3D/System.h]: Oops?

File: src/server/shared/Packets/ByteBuffer.h [hunk 2]
```diff
@@ -393,7 +393,11 @@ class ByteBuffer
             lt.tm_mon = (packedDate >> 20) & 0xF;
             lt.tm_year = ((packedDate >> 24) & 0x1F) + 100;
 
+#ifdef __FreeBSD__
+            return uint32(mktime(&lt));
```
Comment (DDuarte) [inline src/server/shared/Packets/ByteBuffer.h]: Does this change actually fix the build?

**Comment by jackpoz** [comment]
Any change to G3D requires a diff like the ones at https://github.com/TrinityCore/TrinityCore/tree/6.x/dep/g3dlite and a comment in the README specifying the reason of the changes

File: src/tools/mmaps_generator/CMakeLists.txt [hunk 3]
```diff
@@ -46,6 +46,7 @@ target_link_libraries(mmaps_generator
   g3dlib
   Recast
   Detour
+  shared
```
Comment (jackpoz) [inline src/tools/mmaps_generator/CMakeLists.txt]: it's strange that every other platform doesn't require this particular link library, could you please elaborate more about this line ?

File: src/server/shared/Packets/ByteBuffer.h (hunk 2 above)
Comment (DDuarte) [inline src/server/shared/Packets/ByteBuffer.h]: No longer required after Shauren's aa3b05c312626d903228aa7a768a069bce5fe526

File: src/tools/mmaps_generator/CMakeLists.txt (hunk 3 above)
Comment (jackpoz) [inline src/tools/mmaps_generator/CMakeLists.txt]: any update on this ?

**Comment by Carbenium** [comment]
Is this still valid?

**Comment by jackpoz** [comment]
@Carbenium did you read the comments above ?

**Comment by Carbenium** [comment]
@jackpoz: Yes I did. Just wasn't sure after DDuarte's comment.

File: src/tools/mmaps_generator/CMakeLists.txt (hunk 3 above)
Comment (jackpoz) [inline src/tools/mmaps_generator/CMakeLists.txt]: 2nd try: any update on this ?

**Comment by jackpoz** [comment]
this PR hasn't been updated in months by the original author despite all attempts of the TC team.
these changes will be merged (if still required) once a FreeBSD 10.x user will choose to keep TC build on FreeBSD working.

**Pull request closed without merging**